k8s/k8s.go

```diff
@@ -172,23 +172,36 @@ func CmdAddK8s(args *skel.CmdArgs, conf utils.NetConf, hostname string, calicoCl
 		}
 
 		ipAddrsNoIpam := annot["cni.projectcalico.org/ipAddrsNoIpam"]
+		ipAddrs := annot["cni.projectcalico.org/ipAddrs"]
 
-		// Call IPAM plugin if ipAddrsNoIpam annotation is not present.
-		if ipAddrsNoIpam == "" {
-			// Run the IPAM plugin
+		// switch based on which annotations are passed or not passed.
+		switch {
+		case ipAddrs == "" && ipAddrsNoIpam == "":
+			// Call IPAM plugin if ipAddrsNoIpam or ipAddrs annotation is not present.
 			logger.Debugf("Calling IPAM plugin %s", conf.IPAM.Type)
 			result, err = ipam.ExecAdd(conf.IPAM.Type, args.StdinData)
 			if err != nil {
 				return nil, err
 			}
 			logger.Debugf("IPAM plugin returned: %+v", result)
-		} else {
+		case ipAddrs != "" && ipAddrsNoIpam != "":
+			// Can't have both ipAddrs and ipAddrsNoIpam annotations at the same time.
+			logger.Error("Can't have both annotations: 'ipAddrs' and 'ipAddrsNoIpam' in use at the same time")
+			return nil, fmt.Errorf("Can't have both annotations: 'ipAddrs' and 'ipAddrsNoIpam' in use at the same time")
+		case ipAddrsNoIpam != "":
 			// ipAddrsNoIpam annotation is set so bypass IPAM, and set the IPs manually.
 			result, err = overrideIPAMResult(ipAddrsNoIpam, logger)
 			if err != nil {
 				return nil, err
 			}
 			logger.Debugf("Bypassing IPAM to set the result to: %+v", result)
+		case ipAddrs != "":
+			// ipAddrs annotation is set.
```

Comment:
This is a pretty useless comment :)

I think a bit more detail here would be really useful.  e.g "when this annotation is set, we call out to the configured IPAM plugin requesting the specific IP addresses included in the annotation".